synapse/storage/databases/main/room.py

```diff
@@ -1182,14 +1213,7 @@ async def upsert_room_on_join(self, room_id: str, room_version: RoomVersion):
         # It's possible that we already have events for the room in our DB
         # without a corresponding room entry. If we do then we don't want to
         # mark the room as having an auth chain cover index.
```

Comment:
We probably don't need this comment anymore (and the one near line 1267) as it is in the `has_auth_chain_index` method now.

Reply:
I think its a bit unclear why we're bothering to call the function otherwise?

Reply:
I don't feel strongly. I'm OK with leaving it.